aten/src/ATen/native/mps/OperationUtils.mm

```diff
@@ -226,7 +226,7 @@ void printTensorNDArray(const Tensor& t) {
   // extract the pointer to MTLBuffer from the Tensor's storage
   id<MTLBuffer> srcBuf = getMTLBufferStorage(src);
   // a view tensor could be contiguous (e.g., slice ops) or non-contiguous (e.g., transpose())
-  if (src.is_view() || !src.is_contiguous()) {
+  if (src.is_view() || src.storage_offset() != 0 || !src.is_contiguous()) {
```

Comment:
Can you please add a more detailed description, why is this necessary? Shouldn't we just add handling for non-zero storage_offset to the MTLBuffer pointer computation?

Reply:
Let me find if there is a better way to handle this kind of tensor.

Reply:
For now I didn't find a proper way to get an offset of a MTLBuffer (from the [docs](https://developer.apple.com/documentation/metal/mtlbuffer?language=objc) on Appls's website there seems no such methods provided, not sure whether using primitive pointer operations to it is safe or not). What I know is that `MTLComputeCommandEncoder` derived from a `MTLCommandBuffer` in a custom metal kernel can add an offset for reading an existing buffer, like so `[computeEncoder setBuffer:a offset:0 atIndex:0];`, but it seems to not apply to this case. So unfortunately we may have to handle this as is.

Please let me know if I'm missing something.

Reply:
There is a way to add offset by creating Array view. You will have to tile the offset to all the dimensions as its done in the [code path ](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/mps/operations/View.mm#L488).